turbo/snapshotsync/freezeblocks/block_snapshots.go

```diff
@@ -594,7 +593,7 @@ func (s *RoSnapshots) Ranges() []Range {
 func (s *RoSnapshots) OptimisticalyReopenFolder()           { _ = s.ReopenFolder() }
 func (s *RoSnapshots) OptimisticalyReopenWithDB(db kv.RoDB) { _ = s.ReopenWithDB(db) }
 func (s *RoSnapshots) ReopenFolder() error {
-	if err := s.ReopenSegments(s.Types(), false); err != nil {
+	if err := s.ReopenSegments(s.Types(), true); err != nil {
```

Comment:
I still feel it's possible: always allow `[0;N]` gaps (inside noGaps func) and leave this flag `false`. Can you pls add unit-test for noGaps func and allow `[0;N]` gaps?

Reply:
Ok done.